src/code/z_horse.c

```diff
@@ -14,15 +23,23 @@ s32 func_8006CFC0(s32 scene) {
     return 0;
 }
 
-void func_8006D074(GlobalContext* globalCtx) {
+void Horse_SetSpawnInHyruleField(GlobalContext* globalCtx) {
     gSaveContext.horseData.scene = SCENE_SPOT00;
     gSaveContext.horseData.pos.x = -1840;
     gSaveContext.horseData.pos.y = 72;
     gSaveContext.horseData.pos.z = 5497;
     gSaveContext.horseData.angle = -27353;
 }
 
-void func_8006D0AC(GlobalContext* globalCtx) {
+/**
+ * @brief Sets the horses spawn point in Lake Hylia.
+ *
+ * Only works if the horse is located in Lake Hylia.
+ * This is called only when Link obtains the water medallion...
+ *
+ * @param globalCtx
+ */
+void Horse_TrySetSpawnInLakeHylia(GlobalContext* globalCtx) {
```

Comment:
I guess this is to prevent Epona from being in the now filled up lake (so underwater).

The name should probably reflect that this is meant to be called when beating water. Not sure how to phrase it